spec/spec_helper.rb

```diff
@@ -1 +1,23 @@
+if ENV['COVERAGE']
+  # In order to archive an expected coverage analysis we need to eager load
+  # all logstash code base, otherwise it will not get a good analysis.
+
+  require "pluginmanager/main"
+  Dir.glob("lib/logstash/patches/*.rb") do |file|
+    require file
+  end
+
+  Dir.glob("lib/logstash/*.rb") do |file|
+    require file
+  end
+
+  Dir.glob("lib/logstash/util/*.rb") do |file|
+    next if ["lib/logstash/util/prctl.rb"].include?(file)
+    require file
```

Comment:
ok, so it's not related to actual testing/specs but for coverage analysis so I would suggest to create a specific task for eager loading stuff and have the setup-simplecov task depend on that other eager loading task.